src/ast/path_util.rs

```diff
@@ -1,22 +1,33 @@
-use crate::kinds::SyntaxKind::*;
+use crate::{ast::AstToken, kinds::SyntaxKind::*};
 use rowan::{ast::AstNode as OtherAstNode, NodeOrToken};
 
 use crate::ast;
 
-use super::InterpolPart;
+use super::{InterpolPart, PathContent};
 
 impl ast::nodes::Path {
-    pub fn parts(&self) -> impl Iterator<Item = InterpolPart> {
+    pub fn parts(&self) -> impl Iterator<Item = InterpolPart<PathContent>> {
         self.syntax().children_with_tokens().map(|child| match child {
             NodeOrToken::Token(token) => {
                 assert_eq!(token.kind(), TOKEN_PATH);
-                InterpolPart::Literal(token.text().to_string())
+                InterpolPart::Literal(PathContent::cast(token).unwrap())
             }
             NodeOrToken::Node(node) => {
                 InterpolPart::Interpolation(ast::Interpol::cast(node.clone()).unwrap())
             }
         })
     }
+
+    pub fn parts_parsed(&self) -> Vec<InterpolPart<String>> {
+        self.parts()
+            .map(|part| match part {
+                InterpolPart::Literal(literal) => {
+                    InterpolPart::Literal(literal.syntax().text().to_string())
+                }
+                InterpolPart::Interpolation(interpol) => InterpolPart::Interpolation(interpol),
+            })
+            .collect()
```

Comment:
please omit the `.collect()`, what the user intends with the iterator shouldn't be a business of the library, if avoidable.